packages/client/machines/usersUniversalSearcMachine.ts

```diff
@@ -0,0 +1,311 @@
+        tmpAuthUserID: string;
+    }) => Promise<PaginatedUserSummariesSearchResult>;
+}
+
+export function createUsersUniversalSearchMachine(): UsersUniversalSearchMachine {
+    return usersUniversalSearchModel.createMachine(
+        {
+            type: 'parallel',
+
+            states: {
+                steps: {
+                    initial: 'fetchingUsers',
+
+                    states: {
+                        idle: {
+                            on: {
+                                LOAD_MORE_ITEMS: {
+                                    cond: 'hasMoreUsersToFetch',
+                                    target: 'fetchingUsers',
+                                },
+                            },
+                        },
+
+                        fetchingUsers: {
+                            tags: 'fetching',
+
+                            invoke: {
+                                src: 'fetchUsers',
+                            },
+
+                            on: {
+                                FETCHED_USERS: {
+                                    target: 'idle',
+
+                                    actions: assignFetchedUsersToContext,
+                                },
+
+                                FAILED_FETCHING_USERS: {
+                                    target: 'errFetchingUsers',
+                                },
+                            },
+                        },
+
+                        errFetchingUsers: {
+                            tags: 'errorFetchingUsers',
+                        },
+                    },
+
+                    on: {
+                        SUBMITTED: {
+                            cond: (
+                                { searchQuery: currentSearchQuery },
+                                { searchQuery: updatedSearchQuery },
+                            ) => {
+                                const isSameSearchQuery =
+                                    currentSearchQuery === updatedSearchQuery;
+                                const isDifferentSearchQuery =
+                                    isSameSearchQuery === false;
+
+                                return isDifferentSearchQuery;
+                            },
+
+                            target: 'steps.fetchingUsers',
+
+                            actions: assignSearchQueryToContext,
+                        },
+
+                        CLEAR_QUERY: {
+                            target: 'steps.fetchingUsers',
+
+                            actions: usersUniversalSearchModel.assign({
+                                searchQuery: '',
+                                nextPage: 1,
+                            }),
+                        },
+
+                        CANCEL: {
+                            target: 'steps.fetchingUsers',
+
+                            actions: usersUniversalSearchModel.assign({
+                                searchQuery: '',
+                                nextPage: 1,
```

Comment:
Unless I'm mistaken, setting the nextPage to 1 is enough. It will make the machine fetch the selected page, the server result will mutate the `hasMore` boolean depending on it's real value